### PR TITLE
feat: Support exporting Torch-TRT compiled Graphmodules

### PR DESCRIPTION
**Pull request by peri044**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

**Review by narendasan — COMMENTED** [review]
Cant we just reuse the packed format we use for def_pkl?

File: py/torch_tensorrt/dynamo/runtime/register_fake_class.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+@torch.library.register_fake("tensorrt::execute_engine")
+def fake_execute_engine(inputs, trt_engine):
+    breakpoint()
+    return trt_engine(inputs)
+
+
+# namespace::class_name
+@torch._library.register_fake_class("tensorrt::Engine")
+class FakeTRTEngine:
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/runtime/register_fake_class.py:15]: Could we move the TorchTensorRTModule impl here?

File: core/runtime/TRTEngine.cpp [hunk 2]
```diff
@@ -318,6 +318,23 @@ void TRTEngine::verify_serialization_fmt(const std::vector<std::string>& seriali
           << ")");
 }
 
+FlattenedState TRTEngine::__obj_flatten__() {
```
Comment (narendasan) [inline core/runtime/TRTEngine.cpp:321]: Im not sure why this needs to be a method vs reusing the functions in register_jit_hooks

**Comment by narendasan** [comment]
What is the spec for flattened_obj?

**Review by narendasan — APPROVED** [review]
This seems good to me

**Comment by peri044** [comment]
The failing tests are related to `get_hash` for engine cache which is handled in another PR.

**Pull request merged**